extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/GrpcServerRecorder.java

```diff
@@ -226,7 +228,7 @@ private void prodStart(GrpcContainer grpcContainer, Vertx vertx, GrpcServerConfi
 
     private void postStartup(GrpcServerConfiguration configuration, GrpcBuilderProvider<?> provider, boolean test) {
         initHealthStorage();
-        int port = test ? configuration.testPort : configuration.port;
+        int port = test ? testPort(configuration) : configuration.port;
```

Comment:
Well, this was easy. :-)
Are we sure we captured all such log outputs -- with this "new" port?

Reply:
There's an output in `buildServer` but that's before the server is started - so before the port is chosen.